publication/src/main/scala/hmda/publication/reports/disclosure/D51.scala

```diff
@@ -0,0 +1,144 @@
+
+    D51(
+      respondentId,
+      institutionName,
+      "5-1",
+      Disclosure,
+      description,
+      year,
+      reportDate,
+      msa,
+      applicantIncomes,
+      total
+    )
+  }
+
+  val dispositions: List[DispositionType] =
+    List(
+      ReceivedDisp,
+      OriginatedDisp,
+      ApprovedButNotAcceptedDisp,
+      DeniedDisp,
+      WithdrawnDisp,
+      ClosedDisp
+    )
+
+  // Table filters:
+  // Loan Type 2,3,4
+  // Property Type 1,2
+  // Purpose of Loan 1
+  def generate[ec: EC, mat: MAT, as: AS](larSource: Source[LoanApplicationRegisterQuery, NotUsed], fipsCode: Int, respId: String): Future[D51] = {
+
+    val lars = larSource
+      .filter(lar => lar.respondentId == respId)
+      .filter(lar => lar.msa != "NA")
+      .filter(lar => lar.msa.toInt == fipsCode)
+      .filter { lar =>
+        (lar.loanType == 2 || lar.loanType == 3 || lar.loanType == 4) &&
+          (lar.propertyType == 1 || lar.propertyType == 2) &&
+          (lar.purpose == 1)
+      }
+    val larsWithIncome = lars.filter(lar => lar.income != "NA")
+
+    val msa = msaReport(fipsCode.toString)
+
+    val incomeIntervals = calculateMedianIncomeIntervals(fipsCode)
+
+    val larsByIncome = larsByIncomeInterval(larsWithIncome, incomeIntervals)
+    val borrowerCharacteristicsByIncomeF = borrowerCharacteristicsByIncomeInterval(larsByIncome, dispositions)
+
+    val dateF = calculateYear(larSource)
+    val totalF = calculateDispositions(lars, dispositions)
+
+    for {
+      lars50BorrowerCharacteristics <- borrowerCharacteristicsByIncomeF(LessThan50PercentOfMSAMedian)
+      lars50To79BorrowerCharacteristics <- borrowerCharacteristicsByIncomeF(Between50And79PercentOfMSAMedian)
+      lars80To99BorrowerCharacteristics <- borrowerCharacteristicsByIncomeF(Between80And99PercentOfMSAMedian)
+      lars100To120BorrowerCharacteristics <- borrowerCharacteristicsByIncomeF(Between100And119PercentOfMSAMedian)
+      lars120BorrowerCharacteristics <- borrowerCharacteristicsByIncomeF(GreaterThan120PercentOfMSAMedian)
+
+      date <- dateF
+      total <- totalF
+    } yield {
+      val income50 = ApplicantIncome(
+        LessThan50PercentOfMSAMedian,
+        lars50BorrowerCharacteristics
+      )
+      val income50To79 = ApplicantIncome(
+        Between50And79PercentOfMSAMedian,
+        lars50To79BorrowerCharacteristics
+      )
+      val income80To99 = ApplicantIncome(
+        Between80And99PercentOfMSAMedian,
+        lars80To99BorrowerCharacteristics
+      )
+      val income100To120 = ApplicantIncome(
+        Between100And119PercentOfMSAMedian,
+        lars100To120BorrowerCharacteristics
+      )
+      val income120 = ApplicantIncome(
+        GreaterThan120PercentOfMSAMedian,
+        lars120BorrowerCharacteristics
+      )
+
+      D51(
+        respId,
+        "",
```

Comment:
In order to close out this issue, I would say that we should have the full report generated for D5-1 as specified in the ticket.  If we'd still like to add the institution name separately, we should change this PR to "addresses", and create a separate PR to close the issue.